File: my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class Smartphone : public rclcpp::Node
{
public:
    Smartphone() : Node("smartphone")
    {
        subscriber_ = this->create_subscription<example_interfaces::msg::String>("robot_news", 10, std::bind(&Smartphone::robot_news_processing, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "Smartphone Cretaed");
    }

private:
    void robot_news_processing(example_interfaces::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), msg->data.c_str());
    }
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<Smartphone>();
    rclcpp::spin(node);
    rclcpp::shutdown();
}

File: my_py_pkg/my_py_pkg/smartphone.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class SmartphoneNode(Node):
    def __init__(self) -> None: 
        super().__init__("smartphone")
        self.get_logger().info("Created Smartphone Node")
        self.subscriber_ = self.create_subscription(String, "robot_news", self.robot_news_proccessing, 10)
    
    def robot_news_proccessing(self, msg: String) -> None: 
        self.get_logger().info(msg.data)
    

def main() -> None:
    rclpy.init()
    node = SmartphoneNode()
    rclpy.spin(node)
    rclpy.shutdown()